Add debounce for generating swap routes

kkpagaev opened the pull request:
Fix too many request when using inputs in swap form

MattPoblete commented:
This fix actually do more requests than the deployed app (372 vs 231), please fix
<img width="1502" alt="Captura de pantalla 2025-04-29 a la(s) 11 23 30 a m" src="https://github.com/user-attachments/assets/628ceac0-7bc4-48d8-8c36-55f157e82d31" />
<img width="1502" alt="Captura de pantalla 2025-04-29 a la(s) 11 22 19 a m" src="https://github.com/user-attachments/assets/024d8759-7683-47fe-9cc3-646e4e3c9061" />

kkpagaev commented:
@MattPoblete Hi, we have been emailing Esteban on upwork for a few days now and haven't heard back from him.

It would be great if he would reply.

I'm duplicating the last text here:

I just wanted to clarify something, as there might be a small misunderstanding. We’ve resolved the “Too Many Simultaneous Requests” issue as outlined in the assessment document.
However, during the review, a different issue was mentioned. We’re aware of that one too, but it wasn’t part of the specific scope we initially agreed on.

Initially you wanted to see how we work, to evaluate our approach, on the solution of the first task, I would like to get your feedback on this.

We’re ready to move forward and assist further, just let us know if everything is on track and what the next steps are.

Looking forward to your feedback!

esteblock commented:
Also, when forcing an output (exact out), when I change the value
Current situation
![image](https://github.com/user-attachments/assets/20a63f98-c231-4a99-b767-fcd35794e690)


This PR
![image](https://github.com/user-attachments/assets/3d46d5d2-ac49-4e58-8f7a-5add661c4588)

kkpagaev commented:
In this PR, we addressed the issue where:  
- An API request was triggered on every input change.  
- After our changes, this no longer happens because we added a debounce. Now, there's a short delay before the API is called - we wait until the user finishes typing before making the request.

The issues you’ve described in conversation in this PR do not directly relate to our test task:
"Too Many Simultaneous Requests (Inputting values triggers too many RPC/API calls. Causes lag and API)".

1) The fact that the page makes around 4 requests on load - we’ve seen that as well and are ready to work on it within a separate task.  
2) On asset selection (that mentioned here - https://github.com/soroswap/frontend/pull/621#issuecomment-2842359500) is also outside the scope of this particular test task.

Additionally, based on your screenshots, we only see the results of the initial page load - not the behavior when interacting with the input field.

We've prepared two short videos demonstrating the behavior before and after our changes.
In both videos, we entered three zeroes into the swap amount input. You can observe the corresponding network activity in the browser's network tab.

https://github.com/user-attachments/assets/853b4d92-bb19-4cc2-ae95-c919783204ea

This shows the behavior before any changes were applied. As you can see, entering three zeroes triggers three separate requests to the /split endpoint.

https://github.com/user-attachments/assets/62c46b6a-38a9-4714-9f95-340940189e18

This shows the behavior after the changes were implemented. Performing the same action now results in only a single request to the /split endpoint.

Please review our PR again and confirm that the reported issue has indeed been resolved.  

Looking forward to your feedback - thank you!  

@MattPoblete @esteblock
